### DISCUSSION TITLE
Usage with vanilla JS?

### DISCUSSION THREAD

**Discussion by corysimmons** [mdx-js/mdx#1511] [Q&A] [answered]
Is it possible to use MDX with vanilla JS? Every guide I'm finding references something from React.

**Comment by ChristianMurphy** [comment, marked as answer]
Yes, MDX itself is vanilla JS.
It uses a renderer to display to the page, that renderer can be React, Preact, or Vue currently.
All of which can be used as vanilla JS or with JSX.

**Reply by corysimmons** [reply]
Thank you for the quick response!

Is there a usage guide on using MDX without any sort of framework like React/Preact/Vue? And instead just using it with `./index.html` and `./foo.mdx` via Babel and/or Webpack config?

**Reply by ChristianMurphy** [reply]
I think you are conflating two different ideas, working directly with a renderer vs needing a renderer.

You can abstract away the rendering framework so that you don't need to see/think about it.
Parcel does this fairly well https://mdxjs.com/getting-started/parcel

But to render a page, with framework provided compoents, you need a rendering framework :slightly_smiling_face:

**Reply by corysimmons** [reply]
Okay thank you for the help. :)

**Reply by ChristianMurphy** [reply]
And if you don't specifically need JSX support provided by MDX+a framework, the underlying Markdown library https://github.com/micromark/micromark can take markdown directly to HTML
